### DISCUSSION TITLE
having a dedicated option to import Apple Photos

### DISCUSSION THREAD

**Discussion by simulot** [simulot/immich-go#10] [Ideas]
It may have some need for importing photos and albums from an ICloud takeout.

I just have a small archive with few images, no albums, and original photos don't have geotag.
I'll need some inputs to understand what is to be done.

Maybe @zakul can give some inputs.

**Comment by zakul** [comment]
Sure, what do you need?
Example images or some kind of library export?

**Reply by simulot** [reply]
Thanks, 

I have a small example of ICloud photo export. But its owner hasn't enable GPS position and doesn't use albums. So I need some information.

- What are memories? Does it worth to replicate them as album in Immich?
  - I see a bunch of csv file with a bogus name in French "19 décembre 20223.csv"   (seriously Apple!).
  - I see some files named Portraits and a date range (probably in French too)
 
- Albums 
  - How are their presented?
  
- How big archives are split?
  - Do we have files scattered all across the zips?

- Metadata
  - In Photo Details*.csv,  only the date of the picture is interesting.
  - Do GPS location are accessible?

- What about exports from the Photo app on a mac?


From I see, I believe that an import of photos and and album worth to be implemented.

**Comment by zakul** [comment]
Two example images. Both should have metadata and long/latitude.

![IMG_2025](https://github.com/simulot/immich-go/assets/8122545/6f17d530-e8de-4f0e-851c-b0f29e6a530e)
![IMG_3201](https://github.com/simulot/immich-go/assets/8122545/e96d401c-0349-44fb-95a6-e2878af3d62f)

**Reply by simulot** [reply]
Sure, there are GPS in photo's metadata.

How you  have exported the photos? From the Photo App or from the ICloud export?

**Reply by zakul** [reply]
Those two are from an export created by the Photos App. I first imported all my pictures from the phone to my Mac and then created an export of the whole library.
